packages/opentelemetry-metrics/test/Meter.test.ts

```diff
@@ -525,24 +520,173 @@ describe('Meter', () => {
+    it('should pipe through resource', async () => {
+      const valueObserver = meter.createValueObserver('name', {}, result => {
+        result.observe(42, { foo: 'bar' });
+      }) as ValueObserverMetric;
+      assert.ok(valueObserver.resource instanceof Resource);
+
+      const [record] = await valueObserver.getMetricRecord();
+      assert.ok(record.resource instanceof Resource);
+    });
+  });
+
+  describe('#batchObserver', () => {
+    it('should create a batch observer', () => {
+      const measure = meter.createBatchObserver('name', () => {});
+      assert.ok(measure instanceof Metric);
+    });
 
-      observer.setCallback(result => {
-        result.observe(() => 42, { foo: 'bar' });
+    it('should create batch observer with options', () => {
+      const measure = meter.createBatchObserver('name', () => {}, {
+        description: 'desc',
+        unit: '1',
+        disabled: false,
+        maxTimeoutUpdateMS: 100,
       });
+      assert.ok(measure instanceof Metric);
+    });
 
-      const [record] = observer.getMetricRecord();
-      assert.ok(record.resource instanceof Resource);
+    it('should use callback to observe values ', async () => {
+      const tempMetric = meter.createValueObserver('cpu_temp_per_app', {
+        monotonic: false,
+        labelKeys: ['app', 'core'],
+        description: 'desc',
+      }) as ValueObserverMetric;
+
+      const cpuUsageMetric = meter.createValueObserver('cpu_usage_per_app', {
+        monotonic: false,
+        labelKeys: ['app', 'core'],
+        description: 'desc',
+      }) as ValueObserverMetric;
+
+      meter.createBatchObserver(
+        'metric_batch_observer',
+        observerBatchResult => {
+          interface StatItem {
+            usage: number;
+            temp: number;
+          }
+
+          interface Stat {
+            name: string;
+            core1: StatItem;
+            core2: StatItem;
+          }
+
+          function someAsyncMetrics() {
+            return new Promise(resolve => {
+              const stats: Stat[] = [
+                {
+                  name: 'app1',
+                  core1: { usage: 2.1, temp: 67 },
+                  core2: { usage: 3.1, temp: 69 },
+                },
+                {
+                  name: 'app2',
+                  core1: { usage: 1.2, temp: 67 },
+                  core2: { usage: 4.5, temp: 69 },
+                },
+              ];
+              resolve(stats);
+            });
+          }
+
+          Promise.all([
+            someAsyncMetrics(),
+            // simulate waiting
+            new Promise((resolve, reject) => {
+              setTimeout(resolve, 1);
+            }),
+          ]).then((stats: unknown[]) => {
+            const apps = (stats[0] as unknown) as Stat[];
+            apps.forEach(app => {
+              observerBatchResult.observe({ app: app.name, core: '1' }, [
+                tempMetric.observation(app.core1.temp),
```

Comment:
@lykkin I'm not super familiar with the go implementation, but @obecny and I looked through it and from what we saw, it does seem to follow this pattern. Can you point out where you think the difference in the go implementation specifically is?

Reply:
@dyladan Yeah, it looks like the comment I was pointed at was just worded confusingly. The `Observation` struct shouldn't be exposed to users, except in the use of the batch observer. There were some examples of this in the go contrib repo, and in their tests.

This can be marked as resolved, thanks for the explanation!